internal/authentication/service.go

```diff
@@ -0,0 +1,98 @@
+package authentication
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/johan-ag/wishlist/internal/users"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type AuthService interface {
+	Register(registerDTO users.User) (bool, error)
+	Login(loginDTO users.User) (string, error)
+	ValidateToken(token string) (*jwt.Token, error)
+}
+
+func NewAuthService(usersService users.Service) AuthService {
```

Comment:
Se debería retornar el `struct` y no la `interface`.

https://dev.to/lcaparelli/should-my-methods-return-structs-or-interfaces-in-go-3b7